lms/djangoapps/certificates/tests/test_signals.py

```diff
@@ -22,10 +23,22 @@ def setUp(self):
         CertificateGenerationConfiguration.objects.create(enabled=True)
 
     def test_cert_generation_enabled_for_self_paced(self):
-        """ Verify the signal enable the self-generated certificates by default for
+        """
+        Verify the signal enables the self-generated certificates for
         self-paced courses.
         """
         self.assertFalse(certs_api.cert_generation_enabled(self.course.id))
 
-        _listen_for_course_publish('store', self.course.id)
+        _listen_for_course_publish('store', self.course.id, self.course.self_paced)
         self.assertTrue(certs_api.cert_generation_enabled(self.course.id))
+
+    def test_cert_generation_disabled_for_instructor_paced(self):
```

Comment:
Also consider renaming this test accordingly – you might use `test_cert_generation_flag_on_pacing_toggle`